tests/model_registry/model_catalog/test_custom_model_catalog.py

```diff
@@ -0,0 +1,105 @@
+from tests.model_registry.model_catalog.constants import (
+    CUSTOM_ECOSYSTEM_CATALOG,
+    CUSTOM_CATALOG_WITH_FILE,
+    SAMPLE_CATALOG_YAML,
+    EXPECTED_CUSTOM_CATALOG_VALUES,
+    EXPECTED_ECHO_CATALOG_VALUES,
+)
+from ocp_resources.config_map import ConfigMap
+import pytest
+from simple_logger.logger import get_logger
+from typing import Self
+
+from tests.model_registry.model_catalog.utils import execute_get_command
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.mark.parametrize(
+    "updated_catalog_config_map, expected_catalog_values",
+    [
+        pytest.param({"sources_yaml": CUSTOM_ECOSYSTEM_CATALOG}, EXPECTED_ECHO_CATALOG_VALUES, id="rhec_test_catalog"),
+        pytest.param(
+            {"sources_yaml": CUSTOM_CATALOG_WITH_FILE, "sample_yaml": SAMPLE_CATALOG_YAML},
+            EXPECTED_CUSTOM_CATALOG_VALUES,
+            id="file_test_catalog",
+        ),
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures(
+    "updated_dsc_component_state_scope_session",
+    "model_registry_namespace",
+    "model_registry_metadata_db_resources",
+    "model_registry_instance",
+    "updated_catalog_config_map",
+)
+class TestModelCatalogRhec:
+    def test_model_custom_catalog_sources(
+        self: Self,
+        updated_catalog_config_map: tuple[ConfigMap, str, str],
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+        expected_catalog_values: dict[str, str],
+    ):
+        """
+        Validate sources api for model catalog
+        """
+        result = execute_get_command(
+            url=f"{model_catalog_rest_url[0]}sources",
```

Comment:
does `model_catalog_rest_url` contains every time `\` ? should we add a validation to add if not present?

Reply:
Yes it does.